clients/drcachesim/scheduler/scheduler_dynamic.cpp

```diff
@@ -879,8 +879,15 @@ scheduler_dynamic_tmpl_t<RecordType, ReaderType>::rebalance_queues(
                            "Rebalance iteration %d: output %d giving up input %d\n",
                            iteration, i, queue_next->index);
                     inputs_to_add.push_back(queue_next->index);
-                } else
+                } else {
+                    if (status == sched_type_t::STATUS_IDLE) {
```

Comment:
Would it be better if the caller (set_initial_schedule, and others too) understand that idle != error? It seems ok for rebalance_queues to allow the caller to make a distinction between "no unblocked inputs were available" and "rebalancing successful".

Reply:
I think it's better this way as A) this IDLE was just one output's result: so it is not representative and does not mean all the other outputs found nothing; B) there is no current action the caller would take on IDLE vs OK even if the IDLE did mean not a single input was moved (unlike other functions where IDLE is used by the caller to take some action)

Reply:
Elaborating on A: this is the final output's result.  An earlier one returning IDLE is ignored and clobbered with the next output's result.

Though that makes me realize a real error (non-OK, non-IDLE) would also be clobbered: so I added a break in the outer loop for that.